### PR TITLE
Microsoft.Win32.Registry shouldn't be supported on uap 

### PR DESCRIPTION
**Pull request by ViktorHofer**
If we wanna do it with the uap configuration this would be the PR for that.

Fixes https://github.com/dotnet/corefx/issues/15966

cc @danmosemsft @weshaggard

### CODE REVIEW

File: src/System.Diagnostics.Process/src/System.Diagnostics.Process.csproj [hunk 1]
```diff
@@ -6,7 +6,8 @@
     <ProjectGuid>{F55047F8-E47B-46E3-B221-C23595AFE168}</ProjectGuid>
     <RootNamespace>System.Diagnostics.Process</RootNamespace>
     <AssemblyName>System.Diagnostics.Process</AssemblyName>
-    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+	<DefineConstants Condition="'$(TargetGroup)' != 'uap'">$(DefineConstants);FEATURE_REGISTRY</DefineConstants>
```
Comment (danmosemsft) [inline src/System.Diagnostics.Process/src/System.Diagnostics.Process.csproj]: indent

**Comment by danmosemsft** [comment]
Does there need to be another change to remove it from the UAP package? If we know what that is it can go in with this.

**Comment by ViktorHofer** [comment]
CI errors: "System.Configuration.ConfigurationErrorsException : The configuration file has been changed by another program. (/tmp/10vinkkb.3lo/xbql2gsc.spc)". Seems like multiple builds are conflicting here

**Comment by ViktorHofer** [comment]
@dotnet-bot test Innerloop PortableLinux Release x64 Build and Test
@dotnet bot test Innerloop Ubuntu14.04 Release x64 Build and Test

**Comment by danmosemsft** [comment]
Let's find out about the package authoring before merging this.

File: src/System.Diagnostics.Process/src/System.Diagnostics.Process.csproj [hunk 2]
```diff
@@ -6,7 +6,8 @@
     <ProjectGuid>{F55047F8-E47B-46E3-B221-C23595AFE168}</ProjectGuid>
     <RootNamespace>System.Diagnostics.Process</RootNamespace>
     <AssemblyName>System.Diagnostics.Process</AssemblyName>
-    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+	  <DefineConstants Condition="'$(TargetGroup)' != 'uap'">$(DefineConstants);FEATURE_REGISTRY</DefineConstants>
```
Comment (weshaggard) [inline src/System.Diagnostics.Process/src/System.Diagnostics.Process.csproj]: nit: spacing

**Comment by weshaggard** [comment]
>If we wanna do it with the uap configuration this would be the PR for that.

FWIW we already are building a uap configuration which is a unique build configuration https://github.com/ViktorHofer/corefx/blob/master/src/System.Diagnostics.Process/src/Configurations.props#L8. So we would have to do work to not build that configuration.

>Does there need to be another change to remove it from the UAP package? If we know what that is it can go in with this.

The goal isn't to remove Process from the UAP package but to remove Registry. To do that we should remove https://github.com/ViktorHofer/corefx/blob/master/src/Microsoft.Win32.Registry/dir.props#L9.

**Comment by weshaggard** [comment]
Innerloop Ubuntu16.04 arm Debug Cross Build

**Comment by weshaggard** [comment]
@dotnet-bot test Innerloop Ubuntu16.04 arm Debug Cross Build please

**Pull request merged**